### PR TITLE
Add maps to SC2

### PR DESCRIPTION
**Pull request by spacekitteh**
###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by spacekitteh** [comment]
Icky but https://github.com/NixOS/nixpkgs/issues/55631

File: pkgs/applications/science/machine-learning/sc2-headless/maps.nix [hunk 1]
```diff
@@ -7,5 +7,40 @@
...
+  ladder2018season1 = (fetchzip {
+    url = "http://blzdistsc2-a.akamaihd.net/MapPacks/Ladder2018Season1.zip";
+    sha256 = "0p51xj98qg816qm9ywv9zar5llqvqs6bcyns6d5fp2j39fg08v6f";
+    stripRoot = false;
+  }).overrideAttrs(o: { UNZIP = "-P iagreetotheeula";});
+  ladder2018season2 = (fetchzip {
+    url = "http://blzdistsc2-a.akamaihd.net/MapPacks/Ladder2018Season2_Updated.zip";
+    sha256 = "1wjn6vpbymjvjxqf10h7az34fnmhb5dpi878nsydlax25v9lgzqx";
+    stripRoot = false;
+  }).overrideAttrs(o: { UNZIP = "-P iagreetotheeula";});
```
Comment (tilpner) [inline pkgs/applications/science/machine-learning/sc2-headless/maps.nix]: This seems like unnecessary repetition. Consider

```nix
let
  fetchzip' = args: (fetchzip args).overrideAttrs (old: { UNZIP = "-P iagreetotheeula"; });
in {
  ladder2018season2 = fetchzip' { ... };
```

**Review by tilpner — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/machine-learning/sc2-headless/maps.nix (hunk 1 above)
Comment (spacekitteh) [reply pkgs/applications/science/machine-learning/sc2-headless/maps.nix]: Thanks, I couldn't figure out the syntax

File: pkgs/applications/science/machine-learning/sc2-headless/maps.nix [hunk 2]
```diff
@@ -1,11 +1,48 @@
 { fetchzip
 }:
-
+let
+  fetchzip' = args: (fetchzip args).overrideAttrs (old: { UNZIP = "-P iagreetotheeula"; });
```
Comment (grahamc) [inline pkgs/applications/science/machine-learning/sc2-headless/maps.nix]: is the eula the same as the license for sc2?

Comment (spacekitteh) [reply pkgs/applications/science/machine-learning/sc2-headless/maps.nix]: Yep

**Pull request merged**